### ISSUE TITLE
Theme upsell badges are not read by Screen Readers

### ISSUE THREAD

**Issue by youbrokesomething** [Automattic/wp-calypso#75444]
### Quick summary

The upsell badges on /themes aren't keyboard focusable and aren't being read by screen readers. When viewed in the accessibility tree, the badges are under a generic label because they're nested in a span. Free themes also have this problem; screen readers cannot read the 'Free' text because it's in a span without a role or anything to indicate that it is essential.

Example:
```
<span class="theme__upsell">
    <div class="premium-badge theme__upsell-popover theme__marketplace-theme">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="gridicon gridicons-star premium-badge__logo" height="14" width="14">
            <use xlink:href="/calypso/images/../images/gridicons-47c7fb356fcb2d963681.svg#gridicons-star">
            </use>
        </svg>
        <span>Paid</span>
    </div>
</span>
```

While resolving this, it would be a good idea to have the upsell badges link to the plans page or act as an upsell. The badges behave as if they should be clickable but don't perform any action.

### Steps to reproduce

1. On /themes
2. Attempt to focus on an upsell badge via keyboard navigation.

OR

1. On /themes
2. Turn on a screen reader such as VoiceOver
3. Try to listen for the upsell badges

### What you expected to happen

Upsell badges should be navigable via keyboard and read out over a screen reader.

### What actually happened

When using a screen reader, it's as if the badges do not exist.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by taipeicoder** [comment]
Fixed via h #78475
